conan/tools/cmake/toolchain/blocks.py

```diff
@@ -564,8 +564,8 @@ def _get_host_runtime_dirs(self, host_req):
         return host_runtime_dirs
 
     def _join_paths(self, paths):
-        paths = [relativize_path(p, self._conanfile, "${CMAKE_CURRENT_LIST_DIR}") for p in paths]
         paths = [p.replace('\\', '/').replace('$', '\\$').replace('"', '\\"') for p in paths]
+        paths = [relativize_path(p, self._conanfile, "${CMAKE_CURRENT_LIST_DIR}") for p in paths]
```

Comment:
what about handling the escapes inside `relativize_path` (maybe using an argument to tell the function to escape or not?) I can see that there are other uses of the function that also need a escape near the use of `relativize_path`.

Reply:
``relativize_path`` is a generic method for all build systems and other generators as VirtualXXXEnv, it would be a bit ugly to escape CMake strings in it.